src/Symfony/Bridge/Twig/Extension/AssetExtension.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Twig\Extension;
+
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Asset\Packages;
+
+/**
+ * Twig extension for the Symfony Asset component.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class AssetExtension extends \Twig_Extension
+{
+    private $packages;
+
+    public function __construct(Packages $packages)
+    {
+        $this->packages = $packages;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFunctions()
+    {
+        return array(
+            new \Twig_SimpleFunction('asset_path', array($this, 'getAssetPath')),
```

Comment:
The thing with the new name it is that it's a little bit misleading because it can indeed generate absolute url's (not only paths) if you are using an UrlPackage, it just can't generate an absolute url if you are using a PathPackage.